spec/services/pipeline_report_service_spec.rb

```diff
@@ -366,4 +366,67 @@
       expect(JSON.parse(@report)["counts"]["1"]["1"]).not_to include("nr")
     end
   end
+
+  describe "sample contains pathogenic taxa" do
+    before do
+      ResqueSpec.reset!
+
+      @pipeline_run = create(:pipeline_run,
+                             sample: create(:sample, project: create(:project)),
+                             finalized: 1,
+                             total_reads: 100,
+                             adjusted_remaining_reads: 100,
+                             taxon_counts_data: [{
+                               tax_id: 1,
+                               tax_level: 2,
+                               taxon_name: "Escherichia", # nonpathogenic genus
+                               e_value: 0,
+                               genus_taxid: 1,
+                             }, {
+                               tax_id: 2,
+                               tax_level: 1,
+                               taxon_name: "Escherichia albertii", # nonpathogenic species
+                               e_value: 0,
+                               genus_taxid: 1,
+                             }, {
+                               tax_id: 3,
+                               tax_level: 1,
+                               taxon_name: "Escherichia coli", # pathogenic species
+                               e_value: 0,
+                               genus_taxid: 1,
+                             }, {
+                               tax_id: 4,
+                               tax_level: 2,
+                               taxon_name: "Salmonella", # pathogenic genus
+                               e_value: 0,
+                               genus_taxid: 4,
+                             }, {
+                               tax_id: 5,
+                               tax_level: 1,
+                               taxon_name: "Salmonella enterica", # species belonging to pathogenic genus
+                               e_value: 0,
+                               genus_taxid: 4,
+                             },])
+
+      @background = create(:background,
+                           pipeline_run_ids: [
+                             create(:pipeline_run,
+                                    sample: create(:sample, project: create(:project))).id,
+                             create(:pipeline_run,
+                                    sample: create(:sample, project: create(:project))).id,
+                           ])
+      @report = PipelineReportService.call(@pipeline_run, @background.id)
+    end
+
+    it "should tag pathogenic genera correctly" do
```

Comment:
Not to be too picky, but the documentation value of the test is minimal if the documented behavior is "X works correctly". Where are the behaviors we discussed? I can't tell as a reader. For example, all pathogens in a genus should be labeled if the genus is pathogenic.

Reply:
Good point 😅 Thanks for the continuous push towards readability!